rwlib/source/loaders/LoaderTXD.cpp

```diff
@@ -175,13 +175,21 @@ TextureData::Handle createTexture(RW::BSTextureNative& texNative, RW::BinaryStre
     return GL_REPEAT;
   };
 
-	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR);
-	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, glTexFilter(texNative.filterflags));
+	bool useMipMaps = false;
+  if (generateMipMaps) {
+	  glGenerateMipmap(GL_TEXTURE_2D);
+		useMipMaps = true;
+	} else if (hasMipMaps) {
+		// @todo Load mipmaps from file instead
+	  glGenerateMipmap(GL_TEXTURE_2D);
+		useMipMaps = true;
+	}
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
```

Comment:
todo wether this is correct. Might be `glTexFilter(texNative.filterflags, false));` ?